src/opm/output/eclipse/EclipseIO.cpp

```diff
@@ -236,19 +236,11 @@ EclipseIO::EclipseIO( const EclipseState& es,
 {
     if( !this->impl->output_enabled )
         return;
-    {
-        const auto& outputDir = this->impl->outputDir;
 
-        // make sure that the output directory exists, if not try to create it
-        if ( !util_entry_exists( outputDir.c_str() ) ) {
-            util_make_path( outputDir.c_str() );
-        }
+    const auto odir = boost::filesystem::path { this->impl->outputDir };
 
-        if( !util_is_directory( outputDir.c_str() ) ) {
-            throw std::runtime_error( "The path specified as output directory '"
-                                      + outputDir + "' is not a directory");
-        }
-    }
+    if (! boost::filesystem::exists(odir))
+        boost::filesystem::create_directories(odir);
```

Comment:
How will this fare if `odir` already exists as a regular file?

Reply:
It will fail spectacularly.  I don't really know what the appropriate response would be to that.  Rename the file (obviously with a warning)? Stop simulation?